compiler/test/data/typescript/node_modules/interface/constructSignature/singleConstructorWithoutGenerics.d.kt

```diff
@@ -0,0 +1,30 @@
+// [test] singleConstructorWithoutGenerics.module_resolved_name.kt
+@file:Suppress("INTERFACE_WITH_SUPERCLASS", "OVERRIDING_FINAL_MEMBER", "RETURN_TYPE_MISMATCH_ON_OVERRIDE", "CONFLICTING_OVERLOADS")
+
+import kotlin.js.*
+import org.khronos.webgl.*
+import org.w3c.dom.*
+import org.w3c.dom.events.*
+import org.w3c.dom.parsing.*
+import org.w3c.dom.svg.*
+import org.w3c.dom.url.*
+import org.w3c.fetch.*
+import org.w3c.files.*
+import org.w3c.notifications.*
+import org.w3c.performance.*
+import org.w3c.workers.*
+import org.w3c.xhr.*
+
+external interface A {
+    fun foo()
+}
+
+external interface AConstructor {
+    fun bar()
+}
+
+external open class Z : A {
+    companion object : AConstructor
```

Comment:
I found that these Kotlin code is not compiled
Because companion object does not implement abstract members of `AConstructor` and because `B` does not implement abstract member of `Z`
Additionally a bit worried, that user can extend `Z` class and in contructor will be generated something like
```
Z.call(this)
```
But `Z` can not be ready for that

I would suggest to generate in such cases something like
```
external interface A {
    fun foo()
}

external interface AConstructor {
    fun bar()
}

@JsName("Z")
private external interface Z__0 : A {
}

external val Z: AConstructor

external open class B : Z__0 {
    override fun foo()
}
```